src/serac/physics/solid_mechanics.hpp

```diff
@@ -363,6 +372,83 @@ class SolidMechanics<order, dim, Parameters<parameter_space...>, std::integer_se
     bcs_.addEssential(disp_bdr, component_disp_bdr_coef_, displacement_.space(), component);
   }
 
+  /**
+   * @brief Set the displacement essential boundary conditions on a set of true degrees of freedom
+   *
+   * @param true_dofs A set of true degrees of freedom to set the displacement on
+   * @param disp The vector function containing the prescribed displacement values
+   *
+   * The @a true_dofs list can be determined using functions from the @a mfem::ParFiniteElementSpace class.
+   *
+   * @note The coefficient is required to be vector-valued. However, only the dofs specified in the @a true_dofs
+   * array will be set. This means that if the @a true_dofs array only contains dofs for a specific vector component in
+   * a vector-valued finite element space, only that component will be set.
+   */
+  void setDisplacementBCs(const mfem::Array<int>                                          true_dofs,
+                          std::function<void(const mfem::Vector&, double, mfem::Vector&)> disp)
```

Comment:
if I write

```cpp
foo.setDisplacementBCs({1, 2, 3}, bar);
```

it's not immediately clear which overload is being called. 

Does the initializer list in the first argument create a `std::set` (i.e. dirichlet bc by attribute) or an `mfem::Array<int>` (dirichlet b.c. by dof index)?

We could use a more descriptive name (e.g. `setDisplacementBCsByDofList` or something) to disambiguate, although that does make the function name pretty long.

Reply:
I was wondering if this method should even be publicly available at all, but I can imagine times where someone might want to grab a random assortment of dofs they calculated themselves. I agree that at the very least it should have a different name as it is fundamentally different.